### PR TITLE
Add support for pre-shared symmetric key

### PR DESCRIPTION
**Pull request by lucawen**
Its for #13 implemention with option for use or not...

### CODE REVIEW

**Comment by LiveChief** [comment]
Please don't make it optional and default should be yes, for most users and the users that don't need it can turn it off

File: wireguard-install.sh [hunk 1]
```diff
@@ -132,6 +132,16 @@ PublicKey = $SERVER_PUB_KEY
 Endpoint = $ENDPOINT
 AllowedIPs = 0.0.0.0/0,::/0" >> "$HOME/$SERVER_WG_NIC-client.conf"
 
+# Ask for pre-shared symmetric key
+read -r -p "Want to use pre-shared symmetric key? [y/N] " response
```
Comment (angristan) [inline wireguard-install.sh]: It should be at the top of the script with the other `read`.

You can keep a simple if here (so rename your `response` variable.

File: wireguard-install.sh [hunk 2]
```diff
@@ -132,6 +132,16 @@ PublicKey = $SERVER_PUB_KEY
...
+case "$response" in
+    [yY][eE][sS]|[yY]) 
+        CLIENT_SYMM_PRE_KEY=$( wg genpsk )
+        echo "PresharedKey = $CLIENT_SYMM_PRE_KEY" >> "/etc/wireguard/$SERVER_WG_NIC.conf"
+        echo "PresharedKey =$CLIENT_SYMM_PRE_KEY" >> "$HOME/$SERVER_WG_NIC-client.conf"
```
Comment (angristan) [inline wireguard-install.sh]:
```suggestion
        echo "PresharedKey = $CLIENT_SYMM_PRE_KEY" >> "$HOME/$SERVER_WG_NIC-client.conf"
```

**Comment by angristan** [comment]
This is ok for me, we just need to fix some style issues. :)

**Comment by Leopere** [comment]
Why do a symmetric key option what security benefit does this have?

**Comment by LiveChief** [comment]
@Leopere 
https://github.com/angristan/wireguard-install/issues/12

**Comment by lucawen** [comment]
> Please don't make it optional and default should be yes, for most users and the users that don't need it can turn it off
> 
> Also you did it wrong, u never generated the psk key what are u going to forward to the config file?

I generate the psk key inside 'case' in line 139... Its can be optional, because its a 'extra security' not a requirement security, right ? If you dont know what you are doing, you dont do it... So i think its need to be optional, what you think ?

**Comment by Leopere** [comment]
Generally, the highest possible level of security should be the default in my opinion.  The whole point of this is to obfuscate your traffic from your local untrusted end to a remote slightly more trusted end.  At least make the whole thing not be automagically configured with lower tier security than is available.  

Another thing, if we can make high security easily achievable then it's more likely that people will use the more secure options.

**Comment by LiveChief** [comment]
> Generally, the highest possible level of security should be the default in my opinion. The whole point of this is to obfuscate your traffic from your local untrusted end to a remote slightly more trusted end. At least make the whole thing not be automagically configured with lower tier security than is available.
> 
> Another thing, if we can make high security easily achievable then it's more likely that people will use the more secure options.

agreed.

**Comment by lucawen** [comment]
@angristan what you think ?

**Comment by LiveChief** [comment]
> @angristan what you think ?

he wants it optional

**Comment by angristan** [comment]
Optional with the read comment suggesting `y` by default

**Comment by lucawen** [comment]
I think thats it...

**Comment by LiveChief** [comment]
https://github.com/angristan/wireguard-install/pull/19/commits/25f93b8b9db9b391aab068f1a0abf2a64e114096

This is the most stupidest thing i have seen,

Lets say the server has ipv6 but the client only has ipv4, how in the world are you supposed to connect??

OR

Lets say client has ipv6 at home and goes to work tries to use vpn only ipv4 so cannot use vpn.

doesn't make sense

**Comment by Leopere** [comment]
@LiveChief you don't need to overreact because you don't understand what's happening bruv relax.

That PR is only there to skip a question that was there before its already determined by the user themselves earlier on in the script.  There's no point in asking twice.

**Comment by LiveChief** [comment]
> @LiveChief you don't need to overreact because you don't understand what's happening bruv relax.
> 
> That PR is only there to skip a question that was there before its already determined by the user themselves earlier on in the script. There's no point in asking twice.

ah i see sir.

**Comment by angristan** [comment]
@LiveChief Please refrain from commenting here if you don't take the time to read and if you don't know how to behave with people. This is my last warning.

**Pull request merged**